clarity/contracts/alex-vault.clar

```diff
@@ -66,6 +66,14 @@
   )
 )
 
+(define-public (transfer-stx (amount uint) (sender principal) (recipient principal))
+  (begin
+    (try! (check-is-approved sender))
+    (as-contract (unwrap! (stx-transfer? amount tx-sender recipient) ERR-TRANSFER-FAILED))
```

Comment:
The `amount` passed is in 8-digit fixed notation, but `stx-transfer` assumes `amount` passed to it to be in 6-digit fixed notation. Therefore, the `amount` needs to be properly transformed from 8-digit to 6-digit notation.

See:
https://github.com/alexgo-io/alex-v1/blob/505565b57ecfec27fbfa8b7f57c061fba41187d9/clarity/contracts/faucet.clar#L112

Reply:
Ok sure I will look into this calculation

Reply:
Do I need to change this in `mint` function as well?